contributingGuides/STYLE.md

```diff
@@ -737,6 +738,27 @@ const name = user?.name || "default name";
 const name = user?.name ?? "default name";
 ```
 
+## Object Merging: Spread Operator vs. Object.assign
+
+Avoid using the spread operator `({...obj})` in performance-critical situations where object merging occurs frequently (e.g., in loops or repeated function calls). The spread operator creates unnecessary intermediate objects, which can lead to higher memory usage and slower execution.
+
+Instead, consider using `Object.assign()` in these scenarios for better performance. The ESLint `prefer-object-spread` rule has been disabled to allow developers to make informed choices between readability and performance.
+
+- Use the spread operator for simple, small object merges where performance isn't a concern.
+- Use Object.assign() when working with large objects or in performance-sensitive contexts.
+
+```ts
+// BAD: Using the spread operator in a performance-critical loop
```

Comment:
This example is not really clear about the performance aspect. Could we maybe add some context to the example (just on its own it does not seem the problematic). Maybe add it to some loop/ array_reduce